gorm-tools/src/main/groovy/gorm/tools/repository/GormRepo.groovy

```diff
@@ -358,14 +363,9 @@ trait GormRepo<D> implements BulkableRepo<D>, QueryMangoEntityApi<D> {
      * @param args - args passed to delete
      */
     void doRemove(D entity, PersistArgs args) {
-        try {
-            getRepoEventPublisher().doBeforeRemove(this, (GormEntity)entity, args)
-            gormInstanceApi().delete(entity, args as Map)
-            getRepoEventPublisher().doAfterRemove(this, (GormEntity)entity, args)
-        }
-        catch (DataAccessException ex) {
-            throw RepoExceptionSupport.translateException(ex, entity)
```

Comment:
@basejump Unless explicitely flushed, the exception doesnt occur in doRemove
but when transaction gets commited, that resulted in exception never being caught here and RepoExceptionSupport was getting involved in exception handling